src/main/java/com/blackducksoftware/integration/hub/sonar/HubSensor.java

```diff
@@ -61,13 +64,21 @@ public void execute(SensorContext context) {
         HubSonarLogger logger = new HubSonarLogger(Loggers.get(context.getClass()));
         SonarManager sonarManager = new SonarManager(context);
         ComponentHelper componentHelper = new ComponentHelper(sonarManager);
-        RestConnection restConnection = createRestConnection(logger, sonarManager);
-        if (restConnection == null) {
+        Optional<BlackDuckServicesFactory> servicesFactoryOptional = createServiceFactory(logger, sonarManager);
+        if (!servicesFactoryOptional.isPresent()) {
             logger.warn("No connection to the Hub server could be established, skipping Black Duck Hub Sensor.");
             return;
         }
-        HubServicesFactory hubServicesFactory = new HubServicesFactory(restConnection);
-        hubServicesFactory.createPhoneHomeDataService().phoneHome(ThirdPartyName.SONARQUBE, context.getSonarQubeVersion().toString(), sonarManager.getHubPluginVersionFromFile("/plugin.properties"));
+        BlackDuckServicesFactory blackDuckServicesFactory = servicesFactoryOptional.get();
```

Comment:
SonarCloud may complain about not having the `.get()` inside of the if statement (obviously, you would have to reverse the logic if this was the case).